sd-validation/test/shema-org-validation-test.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env mocha */
+
+const assert = require('assert');
+const validateJSONLD = require('../');
+
+describe('schema.org validation', () => {
+  it('reports unknown types', async () => {
+    const errors = await validateJSONLD(`{
+      "@context": "http://schema.org",
+      "@type": "Cat"
+    }`);
+
+    assert.equal(errors.length, 1);
+    assert.equal(errors[0].message, 'Unrecognized schema.org type http://schema.org/Cat');
+  });
+
+  it('reports unknown types', async () => {
+    const errors = await validateJSONLD(`{
+      "@context": "http://schema.org",
+      "@type": ["Article", "Dog"]
+    }`);
+
+    assert.equal(errors.length, 1);
+    assert.equal(errors[0].message, 'Unrecognized schema.org type http://schema.org/Dog');
+  });
+
+  it('reports missing required fields', async () => {
+    const errors = await validateJSONLD(`{
+      "@context": "http://schema.org",
+      "@type": "Article"
+    }`);
+
+    assert.equal(errors.length, 7);
+    assert.equal(errors[0].message, 'Missing required property "author"');
+  });
+
+  it('reports missing recommended fields', async () => {
+    const errors = await validateJSONLD(`{
+      "@context": "http://schema.org",
+      "@type": "Article",
+      "author": "Cat",
+      "datePublished": "Oct 29th 2018",
+      "headline": "Human's New Best Friend - Cat",
+      "image": "https://cats.rock/cat.bmp",
+      "publisher": "Cat Magazine"
+    }`);
+
+    assert.equal(errors.length, 2);
+    assert.equal(errors[0].message, 'Missing recommended property "dateModified"');
+  });
+
+  it('reports unexpected fields', async () => {
+    const errors = await validateJSONLD(`{
+      "@context": "http://schema.org",
+      "@type": "Article",
+      "author": "Cat",
+      "datePublished": "Oct 29th 2017",
+      "dateModified": "Oct 29th 2017",
+      "headline": "Human's New Best Friend - Cat",
+      "image": "https://cats.rock/cat.bmp",
+      "publisher": "Cat Magazine",
+      "mainEntityOfPage": "https://cats.rock/magazine.html",
+      "controversial": true
+    }`);
+
+    assert.equal(errors.length, 1);
+    assert.equal(errors[0].message, 'Unexpected property "controversial"');
+  });
+
+  it('passes if non-schema.org context', async () => {
```

Comment:
feels like at the audit-level we'd consider this situation to be equivalent to zero json-ld on the page === `not-applicable`. right?

Reply:
true!